### PR TITLE
Support std::complex for hip-clang

### PR DESCRIPTION
**Pull request by yxsamliu**
This is for supporting TensorFlow

### CODE REVIEW

File: bin/hipcc [hunk 1]
```diff
@@ -114,7 +114,7 @@ if ($HIP_PLATFORM eq "clang") {
     }
     $ROCM_PATH=$ENV{'ROCM_PATH'} // "/opt/rocm";
     $HIPCC="$HIP_CLANG_PATH/clang++";
-    $HIPCXXFLAGS .= "-std=c++11 -I$HIP_PATH/include";
+    $HIPCXXFLAGS .= "-std=c++11 -isystem $HIP_CLANG_PATH/../lib/clang/7.0.0/include -I$HIP_PATH/include";
```
Comment (mangupta) [inline bin/hipcc]: Does this need to be hard-coded to clang 7.0? Is there some way to get this path from some clang config/binary?

Comment (yxsamliu) [reply bin/hipcc]: Fixed. Thanks.

File: tests/src/deviceLib/hipStdComplex.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANNTY OF ANY KIND, EXPRESS OR
+IMPLIED, INNCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANNY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER INN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR INN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
```
Comment (mangupta) [inline tests/src/deviceLib/hipStdComplex.cpp]: The test does not seem to have a HIT BLOCK. So ctest will not pick this up for testing. Or is this test meant to be run manually?

Comment (yxsamliu) [reply tests/src/deviceLib/hipStdComplex.cpp]: Added HIT block. Thanks.

Comment (yxsamliu) [reply tests/src/deviceLib/hipStdComplex.cpp]: Made changes to the test. Now it is passing.

**Pull request merged**